ibazel/log/log.go

```diff
@@ -0,0 +1,62 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+)
+
+type color string
+
+var writer io.Writer = os.Stderr
+var osExit = os.Exit
+var timeNow = time.Now
+
+const (
+	reset color = "\033[0m"
+
+	errorColor color = "\033[23m"
+	fatalColor color = "\033[110m"
+	logColor   color = "\033[96m"
+)
+
+func log(c color, msg string, args ...interface{}) {
+	fmt.Fprintf(writer, "%siBazel [%s]%s: ",
+		c,
+		timeNow().Format(time.Kitchen),
+		reset)
+	fmt.Fprintf(writer, msg, args...)
+	fmt.Fprintf(writer, "\n")
+}
+
+// Error implements Logger.Error.
```

Comment:
Just for my own education is it idiomatic to have comments like this in go? They don't feel like they add value to me so I was wondering why you chose to include them

Reply:
I won't say idiomatic cause I'm a neophyte, but it is a thing I regularly see in code from people who are much better at this than me. In most languages you declare the type you're implementing, so I think this is just to hint to a user that you're actually implementing that interface.